crates/ruff_linter/src/rules/flake8_pyi/rules/any_eq_ne_annotation.rs

```diff
@@ -8,16 +8,20 @@ use crate::checkers::ast::Checker;
 
 /// ## What it does
 /// Checks for `__eq__` and `__ne__` implementations that use `typing.Any` as
-/// the type annotation for the `obj` parameter.
+/// the type annotation for their second parameter.
 ///
 /// ## Why is this bad?
 /// The Python documentation recommends the use of `object` to "indicate that a
-/// value could be any type in a typesafe manner", while `Any` should be used to
-/// "indicate that a value is dynamically typed."
+/// value could be any type in a typesafe manner". `Any`, on the other hand,
+/// should be seen as an "escape hatch when you need to mix dynamically and
+/// statically typed code". Since using `Any` allows you to write highly unsafe
+/// code, you should generally only use `Any` when the semantics of your code
+/// would otherwise be inexpressible to the type checker.
 ///
-/// The semantics of `__eq__` and `__ne__` are such that the `obj` parameter
-/// should be any type, as opposed to a dynamically typed value. Therefore, the
-/// `object` type annotation is more appropriate.
+/// The expectation in Python is that a comparison of two arbitrary objects
+/// using `==` or `!=` should never raise an exception. As such, the methods
+/// implementing these comparison operators -- `__eq__` and `__ne__` -- should
+/// use `object` rather than `Any` for their second parameter.
```

Comment:
This is probably on me but I don't understand the relation between "`==` or` !=` should never raise an exception" and why the method should then use `object` rather than `Any`. How does using `object` guarantee against exceptions (other than that it might catch a few more typing errors)

Reply:
Basically, for any two objects `x` and `y`, `x == y` should _never_ raise an exception -- calling `x.__eq__(y)` should be safe regardless of what the types are. That means that `object` is the better type here, as it's Python's true "top type" from a type theory perspective.

`Any` should be reserved for situations where you either want to allow unsafe behaviour for some reason, or for situations where your code is safe, but the true type that a function accepts is inexpressible. That should never be the case for `__eq__` or `__ne__` methods -- `object` should do just fine in all cases.

Now, how can I put that more succinctly 😅